pip/pip-321.md

```diff
@@ -0,0 +1,241 @@
+
+- Replication clusters configuration:
+  - Set namespace `ns` replication clusters : cluster1 (local cluster)
+  - Set replication clusters when send message1: cluster1, cluster2.
+
+- Expected:
+  - Message1 can replicate between cluster1 and cluster2.
+
+- Actual:
+  - Topic cannot be created at cluster2, and so the message1 can not be replicated to cluster2.
+
+The root cause of these issues is that topics cannot access clusters that are not included in the replication-clusters of the namespace policy. 
+The replication clusters and allowed clusters are different in the definition, and users don’t always want them to keep the same configuration. 
+But in the current implementation, the replication-clusters and allowed-Clusters are all configured by specifying the replication clusters.
+This will make the topic unable to have its replication policies.
+
+To support geo-replication policies at the topic level and the message level, we must make the cluster configuration at the namespace level more clearly.
+Introduce `allowed-clusters` at the namespace level and make `replication-clusters` only the default replication clusters for the topics under the namespace.
+
+# Goals
+
+## In Scope
+
+The namespace will have a more detailed configuration for clusters. Users can use `replication-clusters` and `allowed-clusters` to configure the different behavioral logic of topics in the multiple clusters, rather than just using `replication-clusters` to control all behaviors.
+
+## Out of Scope
+
+This proposal can be used to solve the problem of topic-level and message level geo-replication can not work as expected. It is the initial motivation for this proposal, but this proposal does not involve modifications to geo-replication.
+
+# High Level Design
+A new namespace policy, `allowed_clusters`, should be added. The `allowed_clusters` policy will specify the clusters where topics under this namespace can be created or loaded. The `replication_cluster` indicates the clusters that are used to create a full mesh replication for all topics under this namespace.
+
+When a namespace has the policy with `allowed_clusters` and `replication_clusters`, the topics under this namespace will replicate data to all `replication_clusters` by default. Additionally, they can have their own flexible replication configuration, which should be a subset of the `allowed_clusters`.
+
+If `allowed_clusters` is not set, `replication_cluster` will be used as the default value for `allowed_cluster`.
+
+If neither `allowed_clusters` nor `replication_cluster` are set, topics under this namespace will only be able to publish/subscribe at the local cluster.
+
+Message-level replication is similar to topic-level replication. The replication clusters of a message should be the subset of the `allowed_cluster`, and are the `replication_clusters` configured at the topic level or namespace level by default.
```

Comment:
Which error will users get if the specified clusters of the message are not a subset of the namespace?
And finally, we will also need to add allowed_clusters for a topic. It could not be a scope of this proposal, but it's better to write it down to the `Out of Scope` section.